### PR TITLE
Feat/refactor fastp

### PR DESCRIPTION
**Pull request by ashwini06**
### This PR:
For UMI workflow, 5' adapter sequences need to trimmed prior to umi extract. In this PR, `fastp` rule is splitted into two different rules, where resultant outputs can be either used for balsamic or UMIworkflow.

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/snakemake_rules/quality_control/fastp.rule [hunk 1]
```diff
@@ -25,25 +25,20 @@ if 'quality_trim' in config['QC'].keys():
...
+        read2 = fastq_dir + "{sample}_2.umi_optimized.fastq.gz",
+        json = qc_dir + "fastp/{sample}_fastp_umi.json",
+        html = qc_dir + "fastp/{sample}_fastp_umi.html",
     benchmark:
-             benchmark_dir + "fastp_" + "{sample}_fastp.tsv"
-    singularity: Path(singularity_image, config["bioinfo_tools"].get("fastp") + ".sif").as_posix() 
+        benchmark_dir + "fastp_umi" + "{sample}_fastp.tsv"
+    singularity: 
+        Path(singularity_image, config["bioinfo_tools"].get("fastp") + ".sif").as_posix()
     params:
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/quality_control/fastp.rule:41]: There are unused params here. Please remove them.

File: BALSAMIC/snakemake_rules/quality_control/fastp.rule [hunk 2]
```diff
@@ -64,18 +59,51 @@ fastp \
...
+    params:
+        housekeeper_id = {"id": "{sample}", "tags": "quality-trimmed-fastq"},
+        tmpdir = tmp_dir,
+        fastq_dir = fastq_dir,
+        qc = " ".join(fastp_param_qc),
+        umi = " ".join(fastp_param_umi),
+        adapter = " ".join(fastp_param_adapter),
+        minimum_length = config["QC"]["min_seq_length"],
+        sample_name = "{sample}",
+        conda = config["bioinfo_tools"].get("fastp")
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/quality_control/fastp.rule:94]: Please remove unused params here as well.

**Review by hassanfa — APPROVED** [review]
some parameters are unused, otherwise good to go! 💯

File: BALSAMIC/snakemake_rules/quality_control/fastp.rule (hunk 1 above)
Comment (ashwini06) [reply BALSAMIC/snakemake_rules/quality_control/fastp.rule:41]: Good catch :) Fixed!

**Pull request merged**